tests/phpunit/tests/comment/wpComment.php

```diff
@@ -62,4 +62,45 @@ public function test_get_instance_should_succeed_for_float_that_is_equal_to_post
 
 		$this->assertEquals( 1, $found->comment_ID );
 	}
+
+
+	/**
+	 * @dataProvider data_test_wp_comment_null_byte
+	 * @ticket 52738
+	 */
+	public function test_wp_comment_null_byte_php_7_or_greater( $comment_data, $expected ) {
+		if ( version_compare( PHP_VERSION, '7.0.0', '<' ) ) {
+			$this->markTestSkipped( 'This test can only run on PHP 7.0 or greater due to illegal member variable name.' );
+		}
+
+		$comment = new WP_Comment( $comment_data );
+
+		$this->assertSame( $comment->comment_content, $expected );
+	}
+
+	public function data_test_wp_comment_null_byte() {
```

Comment:
```suggestion
	/**
	 * Data provider.
	 *
	 * @return array[]
	 */
	public function data_test_wp_comment_null_byte() {
```
🔢 Applies elsewhere in the PR.